### DISCUSSION TITLE
Is there a way to get vocabulary (vector representation of all words) of the pre-trained model of word embedding?

### DISCUSSION THREAD

**Discussion by valmirosjunior** [JohnSnowLabs/spark-nlp#9295] [Q&A]
When I make something like:

```
WordEmbeddingsModel().pretrained("glove_6B_300", "xx")
```

Is it possible to make a call for a method to get the whole vocabulary of the model?
Something like a dictionary with all words and their vector representation.

**Comment by wgeul** [comment]
I face the same issue, I want to verify the representativeness of the embeddings by running it through an encoder-decoder model. I need the vocabulary to assess the accuracy.

**Comment by maziyarpanahi** [comment]
There is no public method to make the whole vocab-vector available in WordEmbeddingsModel (specially in Python), we can open a feature request for this.

**Reply by maziyarpanahi** [reply]
will track the progress here: https://github.com/JohnSnowLabs/spark-nlp/issues/13930

**Comment by maziyarpanahi** [comment]
I believe this is doable for all 3 annotators (WordEmbeddingsModel, Doc2VecModel, and Word2VecModel) in Spark NLP 5.1.1 release: https://github.com/JohnSnowLabs/spark-nlp/releases/tag/5.1.1
